sormas-base/pom.xml

```diff
@@ -828,6 +830,31 @@
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-resources-plugin</artifactId>
 			</plugin>
+
+			<plugin>
+				<groupId>com.diffplug.spotless</groupId>
+				<artifactId>spotless-maven-plugin</artifactId>
+				<version>2.6.1</version>
+				<configuration>
+					<ratchetFrom>origin/development</ratchetFrom>
```

Comment:
With this, Spotless will only reformat and check files that have changed compared to origin/development, it also helps keeping the license header date up-to-date.
This however means that people need to have the hzi-braunschweig/SORMAS-Project repository defined as their `origin` remote, i.e. not named as `upstream`, or that they need to keep their own `origin/development` in sync with `upstream/development`.

An alternative would be to remove that line and reformat all files with new license header dates each year. If we decide to reformat everything though, then we can use `$YEAR` in the license-header template; that way, the template itself doesn't need to change, you only have to run `mvn spotless:apply` each New Year's Day and it will update all dates in source files.

See the documentation of the plugin for more information: https://github.com/diffplug/spotless/tree/main/plugin-maven

Reply:
In order to answer this, we need to decide what the process of using this should look like.

Most developers already have the automatic formatting set-up in their IDEs. I wouldn't want them to have to run spotless before committing. Here using spotless should just be an alternative to setting up the IDE.

The big benefit from my point of view is to have this as part of the CI that is run for each PR to automatically detect when something is not formatted correctly.

Does the ratchetFrom work like this for the Github Actions CI builds? If that's the case then we can probably also expect this from all developers.

On the year automatism: From my understanding it would mean that we will only update the files that have been edited, so it would probably be better to just run a full update of the headers at the start of the year.